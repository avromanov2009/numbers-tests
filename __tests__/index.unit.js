import { runTask, task, sumPositiveElementsOfArray } from "../index";

describe("Index testing", () => {
  it("should be return 7 from [1, 2, 3, 4, 5, 6, 8]", () => {
    expect(runTask([1, 2, 3, 4, 5, 6, 8])).toBe(7);
  });

  it("should be return 3 from [5, -1, 2, 1, 4]", () => {
    expect(runTask([5, -1, 2, 1, 4])).toBe(3);
  });

  it("should be return 3 from [1, 2, 4, 5]", () => {
    expect(runTask([1, 2, 4, 5])).toBe(3);
  });

  it("should be return 4 from [1, 2, 4, 5]", () => {
    expect(runTask([1, 2, 4, 5])).toBe(3);
  });

  describe("Sub-module: summ array elements", () => {
    it("should be 15 on [1, 2, 3, 4, 5]", () => {
      expect(sumPositiveElementsOfArray([1, 2, 3, 4, 5])).toBe(15);
    });

    it("should be 15 on [1, 2, 3, 4, 5]", () => {
      expect(sumPositiveElementsOfArray([-1, -3, 5, -3])).toBe(5);
    });

    it("should be 15 on [1, 2, 3, 4, 5]", () => {
      expect(sumPositiveElementsOfArray([-1, -3, -78, -3])).toBe(0);
    });
  });
});
