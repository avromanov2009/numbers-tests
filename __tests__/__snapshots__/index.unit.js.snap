// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index testing Negatives... should throw exception if sum of elements more then safe integer 1`] = `[Function]`;
